src/pages/taskList/TaskListPage/hook.ts

```diff
@@ -0,0 +1,58 @@
+import { ChangeEvent } from 'react';
+import {SelectChangeEvent} from "@mui/material/Select";
+import {useState} from "react";
+import {TaskType} from "../../../models/Task.ts";
+import {tasks} from "../../../utils/mockData.ts";
+import * as React from "react"
+export const useTaskListHook = () => {
+    const [taskList, setTaskList] = useState<TaskType[]>(tasks);
+    const [selectedLabel, setSelectedLabel] = useState<string[]>([]);
+    const [newValue, setNewValue] = React.useState<string>('');
+    const [inputValue, setInputValue] = React.useState('');
+    const handleAutoCompleteSelectedValue = (event: ChangeEvent<HTMLInputElement>, newValue: string) => {
+        event.preventDefault();
+        setNewValue(newValue);
+        filterTasks(newValue?[...selectedLabel, newValue]:[...selectedLabel])
+    };
+    const handleAutoComplitedChange = (event: ChangeEvent<HTMLInputElement>, newInputValue: string) => {
+        event.preventDefault();
+        setInputValue(newInputValue);
+    };
+
+    const filterTasks = (valuesToFilter: string[]) => {
+        console.log(':valuesToFilter: ', valuesToFilter)
+        if (valuesToFilter.length === 0) {
+            setTaskList(tasks)
+        } else {
+            const searchLowercased = valuesToFilter
```

Comment:
Cased

Reply:
only one cycle to the lowercase, just create a new variable inside the filteredTasks operation